core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -834,7 +834,8 @@ private[spark] class BlockManager(
 
     // Either we're storing bytes and we asynchronously started replication, or we're storing
     // values and need to serialize and replicate them now:
-    if (putLevel.replication > 1) {
+    // Should not replicate the block if putBlockStatus is StorageLevel.NONE
```

Comment:
// Should not replicate the block if its StorageLevel is StorageLevel.NONE or putting it to local is failed
is better.